src/modules/feature/room/roomMembers/dialogs/AddExternalPersonDialog.vue

```diff
@@ -0,0 +1,249 @@
+							v-model="lastName"
+							:label="t('pages.rooms.members.dialog.addExternalPerson.label.lastName')"
+							data-testid="invite-external-person-lastname"
+							:rules="[isNonEmptyString(t('pages.rooms.members.dialog.addExternalPerson.label.lastName.error'))]"
+							@keydown.enter.prevent="onConfirmDetails"
+						/>
+					</template>
+				</VForm>
+			</template>
+			<template #actions>
+				<VSpacer />
+				<div class="mr-4 mb-3">
+					<template v-if="step === 'email'">
+						<VBtn
+							ref="cancelButton"
+							class="ms-auto mr-2"
+							:text="t('common.actions.cancel')"
+							data-testid="invite-participant-cancel-btn"
+							@click="onClose"
+						/>
+						<VBtn
+							ref="addButton"
+							class="ms-auto"
+							color="primary"
+							variant="flat"
+							:text="t('pages.rooms.members.dialog.addExternalPerson.button.add')"
+							data-testid="invite-participant-add-email-btn"
+							@click="onConfirmEmail"
+						/>
+					</template>
+					<template v-if="step === 'details'">
+						<VBtn
+							ref="backButton"
+							class="ms-auto mr-2"
+							:text="t('common.actions.back')"
+							data-testid="invite-participant-back-btn"
+							@click="onBackToEmail"
+						/>
+						<VBtn
+							ref="confirmButton"
+							class="ms-auto"
+							color="primary"
+							variant="flat"
+							:text="t('pages.rooms.members.dialog.addExternalPerson.button.invite')"
+							data-testid="invite-participant-confirm-btn"
+							type="submit"
+							@click="onConfirmDetails"
+						/>
+					</template>
+					<template v-if="step === 'error'">
+						<VBtn
+							ref="closeButton"
+							color="secondary"
+							class="ms-auto mr-2"
+							:text="t('common.labels.close')"
+							data-testid="invite-participant-close-btn"
+							@click="onClose"
+						/>
+					</template>
+				</div>
+			</template>
+		</VCard>
+	</VDialog>
+</template>
+
+<script setup lang="ts">
+import { notifyError } from "@data-app";
+import { useEnvConfig } from "@data-env";
+import { useRoomMembersStore } from "@data-room";
+import { InfoAlert } from "@ui-alert";
+import { isNonEmptyString, isValidEmail } from "@util-validators";
+import { computed, ref, watch } from "vue";
+import { useI18n } from "vue-i18n";
+import { useDisplay } from "vuetify";
+import { VBtn, type VCard, VForm, VSpacer, VTextField } from "vuetify/components";
+
+const roomMembersStore = useRoomMembersStore();
+const applicationName = computed(() => useEnvConfig().value.SC_TITLE.replace("Niedersächsische", "Niedersächsischen"));
+
+const isOpen = defineModel({
+	type: Boolean,
+	required: true,
+});
+
+const emit = defineEmits<{
+	(e: "close"): void;
+}>();
+
+const { t } = useI18n();
+const { xs } = useDisplay();
+
+const step = ref<"email" | "details" | "error">("email");
+
+const addExternalPersonForm = ref<VForm>();
+const emailInput = ref<VTextField>();
+const firstNameInput = ref<VTextField>();
+const lastNameInput = ref<VTextField>();
```

Comment:
We could also use `useTemplateRef("<insertRefHere>")` soo vue will automatically infer the type: 


```suggestion
const addExternalPersonForm = useTemplateRef("addExternalPersonForm");
const emailInput = useTemplateRef("emailInput");
const firstNameInput = useTemplateRef("firstNameInput");
const lastNameInput = useTemplateRef("lastNameInput");
```